### ISSUE TITLE
Condensed components ☂️

### ISSUE THREAD

**Issue by pascal-potvin** [carbon-design-system/carbon#6202]
# Acceptance criteria

Systematically support condensed sizes across core components so teams can build more compact and dense layouts without breaking from the system.

### Justification

Many IBM product teams build content-heavy, data-dense applications where screen real estate directly impacts user efficiency. Also, as teams transition legacy UIs to the latest version of Carbon, they can require smaller component heights and denser layouts to accommodate the amount of information users need to see at once. Acquisitions may also face having the need to translate their smaller size components into ours.

Users working with complex pipelines, models, tables, and rules rely on high information visibility to complete tasks effectively. However, density is not consistently supported across Carbon components. This forces teams to implement custom spacing, type, and sizing overrides, increasing maintenance costs and reducing design–development parity.

While Carbon’s geometry assumes web-based patterns like scrolling and pagination, there are power users in certain products that operate in sustained, tool-like workflows where maximum on-screen visibility is essential. Without systematic support for smaller component sizes, teams struggle to meet these functional needs within the system.

### Why this matters
- Speeds up delivery and reduces maintenance for designers and developers
- Preserves user efficiency in complex workflows
- Improves Carbon adoption in data-heavy products

# Phases

#### Audit, find use cases, and make a final list of component sizes needed.
- [x] Audit the design system components for where we can introduce new potential sizes.
- [x] Get real use cases from products teams of where they need smaller component sizes.
- [x] Make final decisions on which new component sizes we need to add to the system.
- [x] Determine which components to start adding sizes too. Which ones are higher and lower priority. 

#### Add 5-10 high priority components sizes to the system

**Dev:**
- **Contextual layout tokens:** This needs to be done first and separately from this workstream.
- **React/WC:** Implement in both frameworks
- **Figma Code Connect:** This will be done at a later date depending on the outcome of how Figma will be structured. 

**Design:** Add to the Figma kit and Website documentation.

> [!Important]
>  Because of v12 priorities, only complete Phase 1 first. Phase 2 will continue after v12 is released.

Phase 1
- Select `xs`
- Search `xs`
- Pagination `xs`
- Text input `xs`
- Password input `xs`
- Dropdown `xs`
- Combobox `xs`
- Multiselect `xs`
- Data table toolbar `xs`
- Data table pagination `xs`
- Contained tabs `sm`
- Line tabs `sm`
- Vertical tabs `lg`, `md` , `sm`

## After v12 is released

Phase 2
- Number input`xs` 
- File uploader `xs` 
- Date picker `xs` 
- Time picker `xs` 
- Slider `xs` 

**Dev:** 
- **Contextual layout tokens:** At this point, this should already be done.
- **React/WC:** Implement in both frameworks
- **Figma Code Connect:** This will be done at a later date depending on the outcome of how Figma will be structured. 

**Design:** 
- Add to the Figma kit and Website documentation.

---

## Resources

➡️ [Monday board](https://ibm.monday.com/boards/18396529558)
➡️ [Workgroup Figma File](https://www.figma.com/design/VX9PtW3oVwmhjUhUDwh2uY/Data-density-components?node-id=3-18328&t=L0sviQQu5S5yTIvC-1)
➡️ [Carbon Figma File](https://www.figma.com/design/HiQlvN02LJPsgSzzk3t1KO/Condensed-components?node-id=284-36636&t=owbWfgbTbkKRLqxL-1)

**Comment by aagonzales** [comment]
@pascal-potvin hello! Yes, we have heard a need for smaller components before and we are open to exploring the possibility of adding them to the library. 

The images in the box file are not loading for me (box is so annoying with images), we'd love to see you usecase more and any product images you have would be great to see. You can slack them to me if need be.

![image](https://user-images.githubusercontent.com/11670886/83897091-b8d24e00-a71a-11ea-8e79-d50ed7cd96bc.png)

**Comment by andreagmann** [comment]
@pascal-potvin the images are also not loading for me. I was wondering if you have a use case for condensed StructuredList, as we have a need for that in Cloud Pak for Data :)

**Comment by pascal-potvin** [comment]
Sorry for the boxnote issues around images. Here are some As-is legacy images
![image](https://user-images.githubusercontent.com/38508793/84187271-ce6bae80-aa5f-11ea-827a-5c1d355659d3.png)
![image](https://user-images.githubusercontent.com/38508793/84187296-d7f51680-aa5f-11ea-8233-b397fb099cc4.png)
![image](https://user-images.githubusercontent.com/38508793/84187315-ddeaf780-aa5f-11ea-93fd-13fc80113ca0.png)
![image](https://user-images.githubusercontent.com/38508793/84187339-e6dbc900-aa5f-11ea-8180-b302ec6ae1d5.png)

-----

_Here are designs with current Carbon components:_
![image](https://user-images.githubusercontent.com/38508793/84187385-f78c3f00-aa5f-11ea-9c40-606efdd5157f.png)
![image](https://user-images.githubusercontent.com/38508793/84187401-fd822000-aa5f-11ea-8c0d-db5a83145c56.png)

_Same designs but with updated condensed components:_
![image](https://user-images.githubusercontent.com/38508793/84187449-0ffc5980-aa60-11ea-93b5-175e9f551ca5.png)
![image](https://user-images.githubusercontent.com/38508793/84187472-1a1e5800-aa60-11ea-8bdb-a9337282c475.png)

**Comment by pascal-potvin** [comment]
Hi! I just wanted to add additional supporting materials as we keep designing our products towards Carbon 10. 

_Examples_
![image](https://user-images.githubusercontent.com/38508793/85319826-2529ad00-b490-11ea-9c94-ab6e7897c7b0.png)
![image](https://user-images.githubusercontent.com/38508793/85319859-31156f00-b490-11ea-9110-8db223e91a39.png)
![image](https://user-images.githubusercontent.com/38508793/85319890-3a9ed700-b490-11ea-83d4-698d5d80b6f0.png)

**Comment by andreagmann** [comment]
Here's an example of where we would need a condensed StructuredList component in CPDS.

<img width="717" alt="Screen Shot 2020-06-22 at 2 33 08 PM" src="https://user-images.githubusercontent.com/14081333/85348206-b4928880-b4af-11ea-9cd3-cd3e45652a0a.png">

**Comment by pascal-potvin** [comment]
There have been a lot of progress since I opened this ticket and I appreciate the efforts by all here :). Would it be possible to evaluate the implementation of 24 px components as well such as dropdowns? Many of the legacy products would actually benefit from this added space. Having that additional flexibility ensures we are not negatively impacting our users experiences

**Comment by andreagmann** [comment]
Information density is a topic that is still coming up in many conversations. What do we need to do to move this along?

**Comment by aagonzales** [comment]
@andrea-gm Here's what we would need to move forward
- List of all components that would need a condensed variant
- Design specs for each 
- We should also convert this issue into an epic and open individual issues for each components.

I see some of exploration in the Sketch file just need to make sure everything is there.

Also a great way to help move this along might be to look for developers to contribute these in a PR. It should be easier than a whole sale new component since its mostly only modifying one factor. 

I can talk to the team about adding this work into our coming sprints as well. Let us know if someone on your side picks up any of these tasks otherwise we'll get them into our backlog.

**Comment by andrea-gm** [comment]
Thanks @aagonzales, I raised it in the UXE agency repo to see if someone there can pick up some of this work. You can keep an eye on that [in this issue](https://github.ibm.com/CDAI-UX-Engineering/uxe-agency-work/issues/100), and I'll also report back.
